### PR TITLE
Update State Transitions to Process Block Roots

### PR DESCRIPTION
**Pull request by nisdas**
This PR will implement these following conditions:

```py

Let previous_block_root be the tree_hash_root of the previous beacon block processed in the chain.
Set state.latest_block_roots[(state.slot - 1) % LATEST_BLOCK_ROOTS_LENGTH] = previous_block_root.

If state.slot % LATEST_BLOCK_ROOTS_LENGTH == 0 append merkle_root(state.latest_block_roots) to state.batched_block_roots.
```

### CODE REVIEW

File: shared/helperutils/helpers.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package helperutils
```
Comment (terenc3t) [inline shared/helperutils/helpers.go]: please change the package name, `helperutils` utility is just too broad. I think `MerkleRoot` could be under `hashutil`, your call : )

File: shared/helperutils/helpers.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+import "github.com/prysmaticlabs/prysm/shared/hashutil"
+
+// MerkleRoot derives the merkle root from a 2d byte array with each element
+// in the outer array signifying the data that is to be represented in the
+// merkle tree.
+func MerkleRoot(values [][]byte) []byte {
```
Comment (terenc3t) [inline shared/helperutils/helpers.go]: We could use some pseudocode here as well : )

**Comment by nisdas** [comment]
@terenc3t Addressed your comments, PTAL

File: shared/hashutil/merkleRoot.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package hashutil
+
+// MerkleRoot derives the merkle root from a 2d byte array with each element
+// in the outer array signifying the data that is to be represented in the
+// merkle tree.
+// Spec:
+//	def merkle_root(values):
+//	o = [0] * len(values) + values
```
Comment (terenc3t) [inline shared/hashutil/merkleRoot.go]: Don't forget to indent : ) like this:
```
//      def merkle_root(values):
//	     o = [0] * len(values) + values
//           for i in range(len(values)-1, 0, -1):
//               o[i] = hash(o[i*2] + o[i*2+1])
```

**Pull request merged**